src/Valet/App.cs

```diff
@@ -74,6 +74,23 @@ public async Task<int> GetVersionAsync()
         return 0;
     }
 
+    public void CheckForUpdates()
+    {
+        try
+        {
+            var latestImageDigest = _dockerService.GetLatestImageDigestAsync(ValetImage, ValetContainerRegistry)?.Result;
+            var currentImageDigest = _dockerService.GetCurrentImageDigestAsync(ValetImage, ValetContainerRegistry).Result;
+            if (latestImageDigest != null && currentImageDigest != null && !latestImageDigest.Equals(currentImageDigest))
```

Comment:
This check assumes that both a "latest local" and a "latest server" image were found and SHAs were returned.  Then, and only then do we do the comparison.  If, for example, the user has explicitly installed an older version by version tag (e.g. `docker pull ghcr.io/valet-customers/valet-cli:0.1.0.13151`) and doesn't have a `:latest` locally, they will not be notified of an available update.  Is this the desired result?

I can see a case where we don't want to notify a user that their version (if an explicit version, and not `latest`) since it's possible that someone isn't on the latest release for a reason (bug, regression, etc.).  However, if someone isn't on latest, it could be a security risk, so maybe we _do_ want to always notify any time we successfully get a remote 'latest' and it doesn't match whatever comes back from `GetCurrentImageDigestAsync()`?

Reply:
I guess we're assuming that `latest` will always be installed?  https://github.com/github/gh-valet/blob/621e914b1347add959e3efbc2dd50c79f719f15d/src/Valet/App.cs#L53

Maybe what I outlined above isn't a practical concern since `gh valet` has `latest` hard coded?

Reply:
I don't think this is a concern for us, because if someone runs `gh valet` without a docker image with the `latest` tag, then when gh-valet does a `docker run --rm ghcr.io/valet-customers/valet-cli:latest <anything>` the latest ghcr.io/valet-customers/valet-cli image will be downloaded (this is what docker does automatically) so that there is a docker image with the latest tag that exists.

If a customer wants to download a specific version of the valet cli, they would have to download it, and manually set the tag for that release to be `latest` (as I noted in 'Hows this tested' portion of the PR description)

Reply:
Based on the hard coded `latest` tag and @luke-engle's comment [here](https://github.com/github/gh-valet/pull/56#issuecomment-1167942768), I think the logic is good on this.  Do you agree @ethanis?

Reply:
Yep, the above discussion, assumptions, and behavior is accurate 😄